app/dlc-suredbits-client/src/main/scala/org/bitcoins/sbclient/Asset.scala

```diff
@@ -0,0 +1,44 @@
+package org.bitcoins.sbclient
+
+sealed abstract class Asset {
+  def toLowerString: String = this.toString.toLowerCase
+  def toUpperString: String = this.toString.toUpperCase
+}
+
+object Asset {
+  case object BTC extends Asset
+  case object ETH extends Asset
+  case object USD extends Asset
+  case object USDT extends Asset
+  case object LTC extends Asset
+  case object BCH extends Asset
+  case object XRP extends Asset
+  case object EOS extends Asset
+  case object EUR extends Asset
+
+  case object DOGE extends Asset
+  case object STR extends Asset
+
+  case class OtherAsset(name: String) extends Asset {
+    override def toString: String = name
+  }
+
+  def fromLowerString(str: String): Asset = {
+    val name = if (str.toLowerCase == "xbt") "btc" else str
+    val assetOpt = all.find(asset => asset.toLowerString == name)
+    assetOpt.getOrElse(OtherAsset(name.toUpperCase))
+  }
+
+  def fromUpperString(str: String): Asset = {
+    val name = if (str.toUpperCase == "XBT") "BTC" else str
+    val assetOpt = all.find(asset => asset.toUpperString == name)
+    assetOpt.getOrElse(OtherAsset(name))
```

Comment:
Do we want it to not throw errors where we don't get an already defined `Asset`? I feel like this could be prone to mistakes.

Reply:
I think it is a nicer interface to have `TradingPair.fromString("BTCWTF")` return `UnsupportedTradingPair(Asset.BTC, OtherAsset("WTF"))` and then have the server give the error (which it does). `OtherAsset` is essentially like a `Left` in an `Either`